### PR TITLE
[Library v1] Extract component for heart button.

### PR DESCRIPTION
**Pull request by szainmehdi**

### CODE REVIEW

File: api/app/Modules/Accounts/Http/api.php [hunk 1]
```diff
@@ -22,6 +22,7 @@
         Route::put('/tracks', [Controllers\LibraryController::class, 'saveTracks'])
             ->middleware(ClearResponseCache::withTags(CacheTags::SAVED_TRACKS));
 
-        Route::delete('/tracks', [Controllers\LibraryController::class, 'removeSavedTracks']);
+        Route::delete('/tracks', [Controllers\LibraryController::class, 'removeSavedTracks'])
+            ->middleware(ClearResponseCache::withTags(CacheTags::SAVED_TRACKS));
```
Comment (shea786) [inline api/app/Modules/Accounts/Http/api.php:26]: How come we are doing this again, i thought this was done already as part of a different PR?

**Review by shea786 — APPROVED** [review]
Just apart from that one comment, i am happy with everything else

File: api/app/Modules/Accounts/Http/api.php (hunk 1 above)
Comment (szainmehdi) [reply api/app/Modules/Accounts/Http/api.php:26]: I forgot to add the middleware to clear the response cache on the `DELETE` endpoint here in the other PR.
